localstack/utils/docker_utils.py

```diff
@@ -98,22 +98,39 @@ def add(self, port, mapped=None, protocol="tcp"):
         mapped = mapped or port
         if isinstance(port, list):
             for i in range(port[1] - port[0] + 1):
-                self.add(port[0] + i, mapped[0] + i)
+                if isinstance(mapped, list):
+                    self.add(port[0] + i, mapped[0] + i)
+                else:
+                    self.add(port[0] + i, mapped)
             return
         if port is None or int(port) <= 0:
             raise Exception("Unable to add mapping for invalid port: %s" % port)
         if self.contains(port):
+            print("port %s already in mappings" % (port))
             return
+        bisected_host_port = None
         for from_range, to_range in self.mappings.items():
             if not self.in_expanded_range(port, from_range):
                 continue
             if not self.in_expanded_range(mapped, to_range):
                 continue
+            if self.in_range(mapped, to_range) and to_range[0] != to_range[1]:
+                bisected_port_index = mapped - to_range[0]
+                bisected_host_port = from_range[0] + bisected_port_index
+                self.bisect_range(mapped, to_range)
+                self.bisect_range(bisected_host_port, from_range)
+                break
```

Comment:
yes, `break` is the correct code here, although `continue` would still work, `break` is more efficient.  Since mappings cannot overlap, this code will only execute if the new mapping is contained within this one.  Therefore, it isn't necessary to run through the rest of the loop.

Reply:
Right, sorry, misread the logic in the loop - thanks for clarifying 👍